### PR TITLE
feat(Modrinth): new presence

### PR DESCRIPTION
**Pull request by itzme1on**
## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

image
image
image

</details>

### CODE REVIEW

File: websites/M/Modrinth/metadata.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+	"description": {
+		"en": "Search and browse thousands of Minecraft mods on Modrinth with instant, accurate search results!"
+	},
+	"url": [
+		"modrinth.com",
+		"docs.modrinth.com",
+		"blog.modrinth.com"
+	],
+	"version": "1.0.0",
+	"logo": "https://i.imgur.com/DHtRMas.png",
```
Comment (Bas950) [inline websites/M/Modrinth/metadata.json:17]: Logo must be 512x512

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/M/Modrinth/presence.ts [hunk 2]
```diff
@@ -0,0 +1,269 @@
+const presence = new Presence({
+		clientId: "1047217921685991462",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: "logo",
```
Comment (theusaf) [inline websites/M/Modrinth/presence.ts:8]: Imgur URLs are preferred over Discord asset names

**Review by theusaf — CHANGES_REQUESTED** [review]

File: websites/M/Modrinth/presence.ts [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+		clientId: "1047217921685991462",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: "https://i.imgur.com/GnMSCG4.png",
+			startTimestamp: browsingTimestamp,
+		},
+		{ pathname, hostname } = document.location,
```
Comment (DarkVillager) [inline websites/M/Modrinth/presence.ts:11]:
```suggestion
		{ pathname, hostname, href } = document.location,
```

**Review by DarkVillager — CHANGES_REQUESTED** [review]

File: websites/M/Modrinth/presence.ts [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+
+	switch (hostname) {
+		case "modrinth.com": {
+			switch (pathSplit[0]) {
+				case "user": {
+					presenceData.details = `Browsing ${pathSplit[1]}'s profile`;
+					presenceData.buttons = [
+						{
+							label: "View Profile",
+							url: `https://modrinth.com/user/${pathSplit[1]}`,
```
Comment (DarkVillager) [inline websites/M/Modrinth/presence.ts:22]:
```suggestion
							url: href,
```

File: websites/M/Modrinth/presence.ts [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+							presenceData.details = "Browsing plugin versions";
+							break;
+						}
+						case "settings": {
+							presenceData.details = "Browsing project settings";
+							break;
+						}
+						default: {
+							presenceData.details = "Browsing plugin description";
+						}
```
Comment (DarkVillager) [inline websites/M/Modrinth/presence.ts:103]:
```suggestion
```

File: websites/M/Modrinth/presence.ts [hunk 6]
```diff
@@ -0,0 +1,269 @@
...
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: "https://i.imgur.com/GnMSCG4.png",
+			startTimestamp: browsingTimestamp,
+		},
+		{ pathname, hostname, href } = document.location,
+		pathSplit = pathname.split("/").filter(x => x);
```
Comment (DarkVillager) [inline websites/M/Modrinth/presence.ts:12]:
```suggestion
		pathSplit = pathname.split("/").filter(x => x),
		active = document.querySelector('[class*="is-active"]')
```

File: websites/M/Modrinth/presence.ts [hunk 7]
```diff
@@ -0,0 +1,269 @@
...
+							break;
+						}
+						case "settings": {
+							presenceData.details = "Browsing project settings";
+							break;
+						}
+						default: {
+							presenceData.details = "Browsing resourcepack description";
+						}
+					}
```
Comment (DarkVillager) [inline websites/M/Modrinth/presence.ts:142]:
```suggestion
```

File: websites/M/Modrinth/presence.ts [hunk 8]
```diff
@@ -0,0 +1,269 @@
...
+						case "settings": {
+							presenceData.details = "Browsing project settings";
+							break;
+						}
+						default: {
+							presenceData.details = "Browsing modpack description";
+						}
+					}
+					break;
+				}
```
Comment (DarkVillager) [inline websites/M/Modrinth/presence.ts:182]:
```suggestion
```

File: websites/M/Modrinth/presence.ts [hunk 9]
```diff
@@ -0,0 +1,269 @@
...
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: "https://i.imgur.com/GnMSCG4.png",
+			startTimestamp: browsingTimestamp,
+		},
+		{ pathname, hostname, href } = document.location,
+		pathSplit = pathname.split("/").filter(x => x);
+
```
Comment (DarkVillager) [inline websites/M/Modrinth/presence.ts:13]: More compact code, for changelog, gallery, etc.
image
```suggestion
if (active?.textContent) presenceData.details = `Browsing ${pathSplit[0]} ${active?.textContent}`
else {
```

File: websites/M/Modrinth/presence.ts [hunk 10]
```diff
@@ -0,0 +1,269 @@
...
+							break;
+						}
+						case "settings": {
+							presenceData.details = "Browsing project settings";
+							break;
+						}
+						default: {
+							presenceData.details = "Browsing mod description";
+						}
+					}
```
Comment (DarkVillager) [inline websites/M/Modrinth/presence.ts:66]:
```suggestion
```

File: websites/M/Modrinth/presence.ts [hunk 11]
```diff
@@ -0,0 +1,269 @@
...
+						url: href,
+					},
+				];
+				presenceData.details = "Reading blog post";
+				presenceData.state = document.querySelector<HTMLHeadingElement>(
+					"h1.post-title__text"
+				).textContent;
+			} else presenceData.details = "Browsing blog";
+		}
+	}
```
Comment (DarkVillager) [inline websites/M/Modrinth/presence.ts:265]:
```suggestion
	}
	}
```

**Review by DarkVillager — CHANGES_REQUESTED** [review]

**Comment by itzme1on** [comment]
idk, i run `yarn format` and prettier, but same error :/

File: websites/M/Modrinth/presence.ts [hunk 12]
```diff
@@ -8,17 +8,17 @@ presence.on("UpdateData", async () => {
...
 
 	switch (hostname) {
 		case "modrinth.com": {
 			if (active?.textContent) {
-presenceData.details = `Browsing ${
+				presenceData.details = `Browsing ${
 					pathSplit[0]
 				} ${active?.textContent.toLowerCase()}`;
-}
+			}
```
Comment (DarkVillager) [inline websites/M/Modrinth/presence.ts:21]: Without the else or returning it won't work and will just go to plugin in the switch statement.
So Option 1 is providing an else
or Option 2:
``		presence.setActivity(presenceData);
		return;``

**Review by DarkVillager — CHANGES_REQUESTED** [review]

File: websites/M/Modrinth/presence.ts (hunk 12 above)
Comment (itzme1on) [reply websites/M/Modrinth/presence.ts:21]: so i need to replace line 185-186 with
```js
	presence.setActivity(presenceData);
	return;
```

Comment (DarkVillager) [reply websites/M/Modrinth/presence.ts:21]: > so i need to replace line 185-186 with
> 
> ```js
> 	presence.setActivity(presenceData);
> 	return;
> ```

No

Comment (DarkVillager) [inline websites/M/Modrinth/presence.ts:21]:
```suggestion
presence.setActivity(presenceData); 
return;
			}
```

**Review by DarkVillager — CHANGES_REQUESTED** [review]

File: websites/M/Modrinth/presence.ts (hunk 12 above)
Comment (itzme1on) [reply websites/M/Modrinth/presence.ts:21]: I think it's a little bit not what we expected
image
image

Comment (DarkVillager) [reply websites/M/Modrinth/presence.ts:21]: > I think it's a little bit not what we expected image image

oh oops.

File: websites/M/Modrinth/presence.ts [hunk 13]
```diff
@@ -0,0 +1,210 @@
...
+						},
+					];
+					break;
+				}
+				case "settings": {
+					presenceData.details = "Browsing profile settings";
+					break;
+				}
+				case "mods": {
+					presenceData.details = "Looking for wonderful mods";
```
Comment (EncryptedDev) [inline websites/M/Modrinth/presence.ts:33]:
```suggestion
					presenceData.details = "Looking for mods";
```

File: websites/M/Modrinth/presence.ts [hunk 14]
```diff
@@ -0,0 +1,210 @@
...
+					presenceData.details = "Looking for a beautiful resourcepack";
+					break;
+				}
+				case "resourcepack": {
+					presenceData.state = `${
+						document.querySelector<HTMLHeadingElement>("h1.title").textContent
+					}`;
+					presenceData.buttons = [
+						{
+							label: "View Resourcepack",
```
Comment (EncryptedDev) [inline websites/M/Modrinth/presence.ts:88]:
```suggestion
							label: "View Resource Pack",
```

File: websites/M/Modrinth/presence.ts [hunk 15]
```diff
@@ -0,0 +1,210 @@
...
+						presenceData.details = `Browsing ${
+							pathSplit[0]
+						} ${active?.textContent.toLowerCase()}`;
+						presence.setActivity(presenceData);
+						return;
+					}
+					break;
+				}
+				case "resourcepacks": {
+					presenceData.details = "Looking for a beautiful resourcepack";
```
Comment (EncryptedDev) [inline websites/M/Modrinth/presence.ts:79]:
```suggestion
					presenceData.details = "Looking for a resource pack";
```

File: websites/M/Modrinth/presence.ts [hunk 16]
```diff
@@ -0,0 +1,210 @@
...
+						presenceData.details = `Browsing ${
+							pathSplit[0]
+						} ${active?.textContent.toLowerCase()}`;
+						presence.setActivity(presenceData);
+						return;
+					}
+					break;
+				}
+				case "plugins": {
+					presenceData.details = "Looking for useful plugins";
```
Comment (EncryptedDev) [inline websites/M/Modrinth/presence.ts:56]:
```suggestion
					presenceData.details = "Looking for plugins";
```

File: websites/M/Modrinth/presence.ts [hunk 17]
```diff
@@ -0,0 +1,210 @@
...
+						presenceData.details = `Browsing ${
+							pathSplit[0]
+						} ${active?.textContent.toLowerCase()}`;
+						presence.setActivity(presenceData);
+						return;
+					}
+					break;
+				}
+				case "modpacks": {
+					presenceData.details = "Looking for a nice modpack";
```
Comment (EncryptedDev) [inline websites/M/Modrinth/presence.ts:102]:
```suggestion
					presenceData.details = "Looking for a mod pack";
```

File: websites/M/Modrinth/presence.ts [hunk 18]
```diff
@@ -0,0 +1,210 @@
...
+					break;
+				}
+				case "modpack": {
+					presenceData.state = `${
+						document.querySelector<HTMLHeadingElement>("h1.title").textContent
+					}`;
+					presenceData.buttons = [
+						{
+							label: "View Modpack",
```
Comment (EncryptedDev) [inline websites/M/Modrinth/presence.ts:111]:
```suggestion
							label: "View Mod Pack",
```

File: websites/M/Modrinth/presence.ts [hunk 19]
```diff
@@ -0,0 +1,210 @@
...
+							url: href,
+						},
+					];
+					if (active?.textContent) {
+						presenceData.details = `Browsing ${
+							pathSplit[0]
+						} ${active?.textContent.toLowerCase()}`;
+						presence.setActivity(presenceData);
```
Comment (EncryptedDev) [inline websites/M/Modrinth/presence.ts:119]: This is only needed at the end

File: websites/M/Modrinth/presence.ts [hunk 20]
```diff
@@ -0,0 +1,210 @@
...
+						return;
+					}
+					break;
+				}
+				case "notifications": {
+					presenceData.details = "Checking notifications";
+					break;
+				}
+				case "dashboard": {
+					presenceData.details = "Looking into dashboard";
```
Comment (EncryptedDev) [inline websites/M/Modrinth/presence.ts:129]: English isn't englishing

File: websites/M/Modrinth/presence.ts [hunk 21]
```diff
@@ -0,0 +1,210 @@
...
+						case "tutorials": {
+							presenceData.details = "Browsing docs tutorials";
+							break;
+						}
+						case "details": {
+							presenceData.details = "Reading about the API";
+							break;
+						}
+						case "modpacks": {
+							presenceData.details = "Reading about modpacks";
```
Comment (EncryptedDev) [inline websites/M/Modrinth/presence.ts:172]:
```suggestion
							presenceData.details = "Reading about mod packs";
```

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

File: websites/M/Modrinth/presence.ts (hunk 18 above)
Comment (itzme1on) [reply websites/M/Modrinth/presence.ts:111]: on the morinth website, the word modpack is written without separation

File: websites/M/Modrinth/presence.ts (hunk 20 above)
Comment (itzme1on) [reply websites/M/Modrinth/presence.ts:129]: I am from Russia, so I do not know how it would be more correct to write

**Review by Bas950 — APPROVED** [review]

**Review by EncryptedDev — APPROVED** [review]

**Pull request merged**